docs/design/backupstoragelocationrepository-design.md

```diff
@@ -0,0 +1,131 @@
+
+* **New Backup Storage Backend Support**: The BSLR does not introduce support for new types of backup storage backends. It relies on existing BSL-compatible storage configurations and does not extend the supported storage types.
+
+* **User-Facing Backup Interfaces**: This design does not provide user-facing interfaces, CLIs, or web consoles for managing backups or repositories. It focuses on the underlying infrastructure and API-level operations.
+
+* **Metrics and Observability Integrations**: While observability is an important consideration for production systems, this design does not currently include Prometheus metrics integration, alerting features, or comprehensive monitoring capabilities.
+
+
+## High-Level Design
+
+The BSLR will use the BSLR controller within OADP to manage the lifecycle of backup storage repositories. It will interface with the Velero repository manager to prepare and maintain Kopia repositories.
+
+## Detailed Design
+
+### BSLR Overview
+
+* **Pointer to Repository**: Each BSLR serves as a reference to a single Kopia repository within a BSL.
+* **Credential Management**: BSL credentials are stored as OpenShift secrets referenced by the BSL. Kopia-specific credentials are used by the BSLR controller and can also be stored as OpenShift secrets.
+* **Kopia Configuration**: BSLR may contain Kopia-specific configuration parameters such as encryption algorithms, compression settings, and other repository-specific options that are not part of the standard BSL configuration.
+
+### BSLR Controller Responsibilities
+
+* **BSL Monitoring**: The controller observes BSLs and ensures a BSLR exists per BSL.
+* **Kopia Repository Management**: For BSLRs not directly tied to a BSL, the controller creates and manages Kopia repositories.
+
+### Controller Architecture
+
+The controller follows standard Kubernetes controller patterns and includes:
+
+* **Reconciler**: `BackupStorageLocationRepositoryReconciler`
+* **RBAC**: Requires full permissions on `backupstoragelocationrepositories`, including status and finalizers.
+
+### Reconciliation Flow
+
+This controller manages BackupStorageLocationRepository (BSLR) objects and ensures that their associated Kopia repositories are correctly initialized and kept in sync with the desired state.
+
+It reacts to changes in either BackupStorageLocation (BSL) or BSLR objects.
+
+#### Reconcile on BSL Changes
+
+_Note: The defult BSLR spec is marked as managed by the OADP controller._
+
+1. When a `BackupStorageLocation` (BSL) changes:
+   - Check if a corresponding default `BackupStorageLocationRepository` (BSLR) exists for the BSL.
+   - If a default BSLR exists:
+     - Ensure it is not marked for deletion.
+     - Determine if the BSLR spec needs to be updated based on changes in the BSL.
```

Comment:
So we would have a default BSLR for every BSL. Could multiple BSL use the same BSLR?